pkg/rc/replication_controller.go

```diff
@@ -242,9 +284,18 @@ func (rc *replicationController) meetDesires() error {
 		}
 	}
 
-	err = rc.checkForIneligible(current, eligible)
+	ineligible := rc.checkForIneligible(current, eligible)
+	if len(ineligible) > 0 {
+		err := rc.transferNodes(ineligible)
+		if err != nil {
+			return err
+		}
+	}
+
+	// transferNodes() may have changed set of current pods so we must refresh
+	current, err = rc.CurrentPods()
```

Comment:
Found a race condition here that I could never reproduce locally but happened in the CI. The node transfer finished before `rc.ensureConsistency` got called so the ineligible node was scheduled again since it was still in `current`. Not sure if that would happen in reality but it happened in the test.

Reply:
ah that's interesting. It's definitely possible for this to happen in practice (although it probably wouldn't be on the same iteration of the main loop as the node transfer was started).

We should probably make sure that `ensureConsistency()` ignores the old node of a node transfer

Reply:
Yeah actually this doesn't solve the problem. The old node could still be unscheduled between when ensureConsistency() is called finishes. I'll fix this

Reply:
I also think it's okay if we don't make another `CurrentPods()` call here at all. We're never going to be able to call it at exactly the right time to detect changes, so I think we just need to make sure that we make safe choices with the initial `CurrentPods()` call at the beginning of the loop, and plan on the next iteration dealing with changes appropriately.

I think as long as we exclude the new node from `eligible` and the old node from `current` as it pertains to `ensureConsistency()` then we're good